src/py/mains/consensus.py

```diff
@@ -138,22 +157,26 @@ def get_seq_data(config, min_cov_aln, min_len_aln):
                 if len(seq) >= min_len_aln:
                     if len(seqs) == 0:
                         seqs.append(seq) #the "seed"
-                        seed_id = l[0]
+                        seed_len = len(seq)
+                        seed_id = read_id
                     if read_id not in read_ids: #avoidng using the same read twice. seed is used again here by design
                         seqs.append(seq)
                         read_ids.add(read_id)
+                        read_cov += len(seq)
             elif l[0] == "+":
-                if len(seqs) >= min_cov_aln:
-                    seqs = seqs[:1] + sorted(seqs[1:], key=lambda x: -len(x))
-                    yield (seqs[:max_n_read], seed_id, config) 
+                if len(seqs) >= min_n_read and read_cov/seed_len >= min_cov_aln:
+                    seqs = get_longest_reads(seqs, max_n_read, max_cov_aln, sort=True)
```

Comment:
Call another function to sort seqs, and pass that to `get_longest_reads()`. Then drop the `sort` parameter.